### PR TITLE
Add collector to service.go

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/5749

Prep for adding a new metric for failed chart-operator helm releases.

### CODE REVIEW

File: service/collector/collector.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+		go func(collectFunc func(ctx context.Context, ch chan<- prometheus.Metric)) {
+			defer wg.Done()
+			collectFunc(ctx, ch)
+		}(collectFunc)
+	}
+
+	wg.Wait()
+
+	c.logger.LogCtx(ctx, "level", "debug", "message", "finished collecting metrics")
```
Comment (kopiczko) [inline service/collector/collector.go]:
```suggestion
	c.logger.LogCtx(ctx, "level", "debug", "message", "collected metrics")
```

Comment (rossf7) [reply service/collector/collector.go]: Good point. Fixed.

**Comment by xh3b4sd** [comment]
When you work on this please check in with SIG Monitoring as well. We have the concept of the collector set and our own collector interface. See https://github.com/giantswarm/aws-operator/blob/master/service/collector/set.go.

**Comment by rossf7** [comment]
@xh3b4sd Ack. I based this on chart-operator but maybe that hasn't been updated to the latest approach.

**Comment by rossf7** [comment]
@xh3b4sd Done. PTAL

File: service/service.go [hunk 2]
```diff
@@ -265,9 +283,12 @@ func New(config Config) (*Service, error) {
 // Boot starts top level service implementation.
 func (s *Service) Boot() {
 	s.bootOnce.Do(func() {
-		go s.AWSClusterController.Boot(context.Background())
-		go s.AzureClusterController.Boot(context.Background())
-		go s.KVMClusterController.Boot(context.Background())
+		prometheus.MustRegister(s.metricsCollector)
```
Comment (xh3b4sd) [inline service/service.go]: The collector should be booted, which does the registration. You can see this in the other operators. Also see https://github.com/giantswarm/exporterkit/blob/master/collector/set.go.

Comment (rossf7) [reply service/service.go]: Yes sorry I missed some of the wiring. Getting it fixed.

**Comment by rossf7** [comment]
Tested on `ginger`.

OK now this should be good to go. @xh3b4sd PTAL

**Pull request merged**